plugins/tpm/fu-tpm-device.c

```diff
@@ -0,0 +1,266 @@
+/*
+ * Copyright (C) 2019 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <tss2/tss2_esys.h>
+
+#include "fu-tpm-device.h"
+
+struct _FuTpmDevice {
+	FuUdevDevice		 parent_instance;
+};
+
+G_DEFINE_TYPE (FuTpmDevice, fu_tpm_device, FU_TYPE_UDEV_DEVICE)
+
+static void Esys_Finalize_autoptr_cleanup (ESYS_CONTEXT *esys_context)
+{
+	Esys_Finalize (&esys_context);
+}
+G_DEFINE_AUTOPTR_CLEANUP_FUNC (ESYS_CONTEXT, Esys_Finalize_autoptr_cleanup)
+
+static gboolean
+fu_tpm_device_probe (FuUdevDevice *device, GError **error)
+{
+	FuTpmDevice *self = FU_TPM_DEVICE (device);
+
+	/* set the physical ID */
+	if (!fu_udev_device_set_physical_id (device, "tpm", error))
+		return FALSE;
+
+	/* already open, so don't touch */
+	g_object_set (self, "device-file", NULL, NULL);
```

Comment:
This feels a little hacky. Maybe it's better to have a proper flag.

Reply:
Agree. Give me a couple of mins.

Reply:
https://github.com/fwupd/fwupd/pull/1643

Reply:
Fixed.